lib/util.ex

```diff
@@ -67,27 +72,42 @@ defmodule Util do
       Authorization: "Bearer " <> auth
     ]
 
-    {:ok, resp} = HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers)
+    case HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers) do
+      {:ok, resp} ->
+        resp.body
+        |> JSON.decode
+        |> format_decoded_response
+      {:error, reason} ->
+        IO.puts(:stderr, reason)
+        System.stop(1)
+    end
+  end
 
-    resp.body
-    |> JSON.decode!()
-    |> Map.get("data")
-    |> Map.get("tournaments")
-    |> Map.get("nodes")
-    # Finesse data into our own non-start.gg format.
-    |> Enum.map(fn sgg_data ->
-      %{
-        "external_id" => "smashgg-" <> Integer.to_string(sgg_data["id"]),
-        "name" => sgg_data["name"],
-        "location" => %{
-          "lat" => sgg_data["lat"],
-          "lng" => sgg_data["lng"]
-        },
-        "start_time" => sgg_data["startAt"],
-        "end_time" => sgg_data["endAt"],
-        "updated_at" => sgg_data["updatedAt"],
-        "url" => "https://start.gg/" <> (sgg_data["slug"])
-      }
-    end)
+  def format_decoded_response({:ok, decoded_map}) do
+    case {:ok, decoded_map} do
+      {:ok, decoded_map = %{"success" => false}} ->
+        IO.puts(:stderr, Map.get(decoded_map, "message"))
+        System.stop(1)
+      {:ok, decoded_map} ->
+        decoded_map
+        |> Map.get("data")
+        |> Map.get("tournaments")
+        |> Map.get("nodes")
+        # Finesse data into our own non-start.gg format.
+        |> Enum.map(fn sgg_data ->
+          %{
+            "external_id" => "smashgg-" <> Integer.to_string(sgg_data["id"]),
+            "name" => sgg_data["name"],
+            "location" => %{
+              "lat" => sgg_data["lat"],
+              "lng" => sgg_data["lng"]
+            },
+            "start_time" => sgg_data["startAt"],
+            "end_time" => sgg_data["endAt"],
+            "updated_at" => sgg_data["updatedAt"],
+            "url" => "https://start.gg/" <> (sgg_data["slug"])
+          }
+          end )
```

Comment:
Figured this is clear enough of a broken out function. Of course we could write two separate format_decoded_response functions to further refactor this, so definitely open to taking it one step further that way if need be.

Reply:
Since there's no logic outside the case here, idiomatic elixir would definitely be to break this into multiple functions, one matching on `{:ok, decoded_map = %{"success" => false}}`, the other matching on `{:ok, decoded_map}`.

That said, if you run into any difficulties, I'm fine committing this and refactoring later lol.